dotCMS/src/main/java/com/dotmarketing/portlets/templates/action/EditTemplateAction.java

```diff
@@ -311,33 +326,30 @@ else if ((cmd != null) && cmd.equals(com.dotmarketing.util.Constants.FULL_DELETE
 			{
 				Logger.debug(this,"Calling Full Delete Method");
 				String [] inodes = req.getParameterValues("publishInode");
-				StringBuilder dependencies = new StringBuilder();
-				boolean returnValue = false;
 
+				int errorCount =0;
 				for(String inode  : inodes)	{
-					String result = null;
-
 					WebAsset webAsset = (WebAsset) InodeFactory.getInode(inode,Template.class);
+					List<Contentlet> pages = APILocator.getContentletAPI().findPagesByTemplate((Template)webAsset, user, false);
 
-					if(UtilMethods.isSet(result)) {
-						dependencies.append(LanguageUtil.get(user, "template-name")).append(": ").append(webAsset.getFriendlyName()).append("\n");
-						dependencies.append(LanguageUtil.get(user, "Pages-URLs")).append(": ").append(result);
+					if(pages!= null && !pages.isEmpty()) {
+						StringBuilder error = new StringBuilder();
+						error.append(LanguageUtil.get(user, "message.template.full_delete.error")).append(" ");
+						error.append(webAsset.getName()).append("<br>");
+
+						for (Contentlet page : pages) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)